app/views/document/show.html.erb

```diff
@@ -0,0 +1,31 @@
+<div class="container">
+  <div>
+    <h2>
+      <% if @user %>
+        Shared file from <b><%= @user.username %></b>
+      <% else %>
+        File not available
+      <% end %>
+    </h2>
+  </div>
+  <% if @document %>
+    <table class="table table-bordered">
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Upload Date</th>
+          <th>Size</th>
+          <th>Action</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td><%= @document.user_document.filename %></td>
+          <td><%= @document.user_document.created_at.strftime("%-d %b %Y %H:%M") %></td>
+          <td><%= human_readable_size @document.user_document.byte_size %></td>
+          <td><%= link_to 'Download', @document.user_document, download: '' %></td>
```

Comment:
A download link of this format can be shared to anyone and even without auth user can download a document. Please update it so that either user auth is required or document should be shared.

Basically on a shared doc if someone copies download link, if document is unshared, link should not work anymore for download.

Reply:
This has been taken care, when user click on shared toggle button Link gets enabled and can be shared with non auth users as well.